src/librustc_trait_selection/traits/wf.rs

```diff
@@ -93,41 +93,45 @@ pub fn predicate_obligations<'a, 'tcx>(
 ) -> Vec<traits::PredicateObligation<'tcx>> {
     let mut wf = WfPredicates { infcx, param_env, body_id, span, out: vec![], item: None };
 
-    // (*) ok to skip binders, because wf code is prepared for it
     match predicate.kind() {
-        ty::PredicateKind::Trait(t, _) => {
-            wf.compute_trait_ref(&t.skip_binder().trait_ref, Elaborate::None); // (*)
+        ty::PredicateKind::ForAll(binder) => {
```

Comment:
might read more nicely if this was `match predicate.skip_binders()` and put the comment on top of that